api/api.go

```diff
@@ -108,6 +111,69 @@ func NewApi(clientset kubernetes.Interface, fasitUrl, clusterDomain, clusterName
 	}
 }
 
+func (api Api) deploy(w http.ResponseWriter, r *http.Request) *appError {
+	requests.With(prometheus.Labels{"path": "deploy"}).Inc()
+
+	deploymentRequest, err := unmarshalDeploymentRequest(r.Body)
+
+	if err != nil {
+		return &appError{err, "unable to unmarshal deployment request", http.StatusBadRequest}
+	}
+
+	//TODO remove this once grace period ends
+	deploymentRequest, warnings := ensurePropertyCompatability(deploymentRequest)
+
+	fasit := FasitClient{api.FasitUrl, deploymentRequest.Username, deploymentRequest.Password}
```

Comment:
Shouldn't we check for which `username` and `password` we should use?

Reply:
?